ports/sonic.1/Sonic 1.sh

```diff
@@ -13,29 +13,86 @@ else
 fi
 
 source $controlfolder/control.txt
-source $controlfolder/device_info.txt
+get_controls
 
+# Source Device Info
+source $controlfolder/device_info.txt
 [ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
 
-get_controls
+# Set variables
+GAMEDIR="/$directory/ports/sonic.1"
+> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
+
+# Set current virtual screen
+if [ "$CFW_NAME" == "muOS" ]; then
+  /opt/muos/extra/muxlog & CUR_TTY="/tmp/muxlog_info"
+elif [ "$CFW_NAME" == "TrimUI" ]; then
+  CUR_TTY="/dev/fd/1"
+else
+  CUR_TTY="/dev/tty0"
+fi
+
+cd $GAMEDIR
 
+# Exports
+LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$GAMEDIR/libs"
+
+# Setup gl4es environment
+if [ -f "${controlfolder}/libgl_${CFW_NAME}.txt" ]; then 
+  source "${controlfolder}/libgl_${CFW_NAME}.txt"
+else
+  source "${controlfolder}/libgl_default.txt"
+fi
+
+# Permissions
+$ESUDO chmod 666 /dev/tty0
 $ESUDO chmod 666 /dev/tty1
+$ESUDO chmod 777 $GAMEDIR/sonic2013
+$ESUDO chmod 777 $GAMEDIR/sonicforever
 
-GAMEDIR="/$directory/ports/sonic1"
-> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
+# Modify ScreenWidth
+LOW=214 # 3:2
+MED=320 # 4:3
+HIGH=426 # 16:9
 
-cd $GAMEDIR
+# Set WIDTH based on DISPLAY_WIDTH
+case $DISPLAY_WIDTH in
+  [0-2][0-9][0-9])  # 0 to 299 range
+    WIDTH=$LOW
+    ;;
+  [3-9][0-9][0-9])  # 300 to 999 range
+    WIDTH=$MED
+    ;;
+  [1-9][0-9][0-9][0-9])  # 1000 and above range
+    WIDTH=$HIGH
+    ;;
+  *)
+    echo "Unknown screen width: $DISPLAY_WIDTH"
+    WIDTH=$MED  # Default value or handle as needed
+    ;;
+esac
 
-export LD_LIBRARY_PATH="$GAMEDIR/libs:$LD_LIBRARY_PATH"
-export SDL_GAMECONTROLLERCONFIG="$sdl_controllerconfig"
+if grep -q "^ScreenWidth=[0-9]\+" "$GAMEDIR/settings.ini"; then
+    sed -i "s/^ScreenWidth=[0-9]\+/ScreenWidth=$WIDTH/" "$GAMEDIR/settings.ini"
+else
+    echo "Possible invalid or missing settings.ini!"
+fi
 
-if [ -n "$(pgrep sway)" ]; then
-  timeout 7 watch swaymsg '[app_id=sonic2013] fullscreen enable' &
```

Comment:
Missing the swaymsg to enable fullscreen on ROCKNIX?

Reply:
There is no `fi` at the end of check.